### PR TITLE
refactoring: update http module to http client

### LINKED ISSUE #7: Upgrade to new angular http client
This will be a braking change.

### PR DESCRIPTION
**Pull request by rokerkony**
**Breaking changes**
- Uses new Http client.
- Headers are now key=>value and built by api client.
- Typescript 2.4.1 or higher required.
- @angular/common 2.3.0 or higher required.

Closes #7

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -97,7 +96,7 @@ of required retries and the ApiClient will take care of it.
 
 - __urlPathParameters__
 
-  This is required when the wildcard is included in the url.
+  This value is required when the wildcard is included in the url.
   
   Example: `{role: 'admin'}` with the url: `/user/:role` will generate: `/user/role`
```
Comment (FlorianTopf) [inline README.md]: will generate `/user/admin`...

File: package.json [hunk 2]
```diff
@@ -36,8 +36,9 @@
   "license": "MIT",
   "peerDependencies": {
     "@angular/core": ">=4.0.0",
```
Comment (FlorianTopf) [inline package.json]: I would raise this to 4.3.0.

File: README.md (hunk 1 above)
Comment (rokerkony) [reply README.md]: 👍

File: package.json (hunk 2 above)
Comment (rokerkony) [reply package.json]: agree, Angular keeps the versions always together... 👍

**Pull request merged**